.clang-tidy

```diff
@@ -25,11 +25,23 @@ Checks: >
   abseil-*,
   bugprone-*,
   -bugprone-macro-parentheses,
+  -bugprone-easily-swappable-parameters,
+  -bugprone-narrowing-conversions,
+  -bugprone-implicit-widening-of-multiplication-result,
+  cert-*,
+  cppcoreguidelines-*,
+  -cppcoreguidelines-non-private-member-variables-in-classes,
+  -cppcoreguidelines-narrowing-conversions,
+  -cppcoreguidelines-avoid-magic-numbers,
```

Comment:
Curious why do we disable this one?

Reply:
we generally have a lot of "magic numbering" all over, imo not a specific important thing to fix atm, open to adding it back

Reply:
it's not a problem: we shouldn't block CI breakage or PR merge for clang-tidy.